lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureLocation.java

```diff
@@ -105,7 +106,7 @@ public String account()
 
     public String path()
     {
-        return location.path();
+        return Utility.urlEncode(location.path());
```

Comment:
you're changing the semantics of `AzureLocation.path()` method and it no longer matches `Location.path()`, which can have undesired consequences (in the code existing now, or in code written in the future)

rename to `encodedPath` (or add encodedPath as a new method)

Reply:
I think @nineinchnick  is on the right path - likely we need a separate method `encodedPath()` here besides the existing `path()`

Reply:
I moved `urlEncode` to `AzureFilesystem`.